Add main mlpack documentation framework (#3603)
mlpack/mlpack: doc/index.md

# Documentation for mlpack

## A fast, flexible machine learning library

mlpack is an intuitive, fast, and flexible header-only C++ machine learning
library with bindings to other languages.  It aims to provide fast, lightweight
implementations of both common and cutting-edge machine learning algorithms.

> mlpack's lightweight C++ implementation makes it ideal for deployment, and it

Reviewer: ```suggestion
mlpack's [lightweight C++ implementation](quickstart/cpp.md) makes it ideal for deployment, and it
```

Author: I did it in a slightly different place (closer to the binding quickstart links) in 680fee01be ---thanks for the suggestion. :+1:

can also be used for interactive prototyping via C++ notebooks (these can be
seen in action on mlpack's [homepage](https://www.mlpack.org/)).

In addition to its powerful C++ interface, mlpack also provides command-line

> programs, and bindings to the Python, R, Julia, and Go languages.

Reviewer: ```suggestion
programs, and bindings to the [Python](quickstart/python.md), [R](quickstart/r.md), [Julia](quickstart/julia.md), and [Go](quickstart/go.md) languages.
```

Add links to get started. 

Maybe list shell/CLI here too?

_If you use mlpack, please [cite the software](citation.md)._

## mlpack basics

Installing mlpack can be done using the
[instructions in the README](README.md#3-installing-and-using-mlpack-in-c);
or the [Windows build guide](user/build_windows.md).
The following basic guides are *highly recommended* before using mlpack.

 * ***First steps***:
   - [mlpack C++ quickstart](quickstart/cpp.md): create a couple simple C++
     programs that use mlpack
   - [Sample Windows mlpack C++ application](user/sample_ml_app.md): create a
     working mlpack Windows program using Visual Studio

 * ***Basics of matrices and data in mlpack***:
   - [Matrices and data in mlpack](user/matrices.md)
   - [Loading and saving mlpack objects](user/load_save.md)

 * ***Reference for mlpack core classes***:
   - [Core mlpack documentation](user/core.md)

>  * ***Bindings to other languages*** (Python/R/CLI/Julia/Go):

Reviewer: ```suggestion
 * ***Using mlpack natively with our extensions in Python, R, CLI, Julia, and Go *** :
```

Bindings is probably a bit too abstract. This emphasizes more an enhancement.

Author: Thanks, this is a nice improvement. :+1:
